redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/TreehouseApp.kt

```diff
@@ -62,7 +71,7 @@ public class TreehouseApp<A : AppService> private constructor(
    * instance may be replaced if new code is loaded.
    */
   public val zipline: Zipline?
-    get() = codeHost.session?.zipline
+    get() = (codeHost.codeSession as? ZiplineCodeSession)?.zipline
```

Comment:
Is `codeSession` not being a `ZiplineCodeSession` valid?

If not, perhaps we can make `CodeSession` itself a type parameter?

Reply:
Lemme explore that in follow up!